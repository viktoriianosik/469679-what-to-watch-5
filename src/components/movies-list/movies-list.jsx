import React from "react";
import PropTypes from "prop-types";
import MovieCard from "../movie-card/movie-card";
import MoviePropTypes from "../movie/movie-props";
import ShowMore from "../show-more/show-more";
import {connect} from "react-redux";
import {ActionCreator} from "../../store/action";
import {MOVIES_COUNT} from "../../const";

const MoviesList = (props) => {
  const {
    movies,
    onMouseEnterCard,
    onMouseLeaveCard,
    onCardClick,
    activeMovie,
    moviesCount,
    onMoreButtonClick,
  } = props;

  return (
    <React.Fragment>
      <div className="catalog__movies-list">
        {
          movies.slice(0, moviesCount).map((movie, i) => (
            <MovieCard
              key={`movie ${i}`}
              movie={movie}
              onMouseEnterCard={() => {
                onMouseEnterCard(movie.id);
              }}
              onMouseLeaveCard={onMouseLeaveCard}
              onCardClick={() => {
                onCardClick(movie.id);
              }}
              isPlaying={movie.id === activeMovie}
            />
          ))
        }
      </div>
      {movies.length > moviesCount && <ShowMore onMoreButtonClick={() => {
        onMoreButtonClick(moviesCount + MOVIES_COUNT);
      }}/>}
    </React.Fragment>
  );
};

MoviesList.propTypes = {
  movies: PropTypes.arrayOf(MoviePropTypes).isRequired,
  activeMovie: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  onMouseEnterCard: PropTypes.func.isRequired,
  onMouseLeaveCard: PropTypes.func.isRequired,
  onCardClick: PropTypes.func.isRequired,
  onMoreButtonClick: PropTypes.func.isRequired,
  moviesCount: PropTypes.number.isRequired,
};

const mapStateToProps = ({PROCESS}) => ({
  moviesCount: PROCESS.moviesCount
});

const mapDispatchToProps = (dispatch) => ({
  onMoreButtonClick(moviesCount) {
    dispatch(ActionCreator.changeMoviesCountList(moviesCount));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(MoviesList);
