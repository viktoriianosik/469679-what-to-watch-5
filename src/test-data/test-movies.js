export const testMovies = [
  {
    backgroundColor: `#795433`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    genre: `Crime`,
    id: 1,
    isFavorite: false,
    name: `Pulp Fiction`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 1.5,
    released: 1994,
    runTime: 153,
    scoresCount: 1635992,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  },
  {
    backgroundColor: `#BDAD8F`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/No_Country_for_Old_Men.jpg`,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    director: `Ethan Coen`,
    genre: `Crime`,
    id: 14,
    isFavorite: false,
    name: `No Country for Old Men`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/No_Country_for_Old_Men.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/no-country-for-old-men.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 4.1,
    released: 2007,
    runTime: 122,
    scoresCount: 764976,
    starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  }
];
