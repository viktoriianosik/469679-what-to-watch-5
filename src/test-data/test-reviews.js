export const testReviews = [
  {
    comment: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2020-10-09T13:38:44.769Z`,
    id: 1,
    rating: 3.3,
    user: {
      id: 19,
      name: `Christina`
    },
  },
  {
    comment: `I love this movie.`,
    date: `2020-13-09T08:38:44.769Z`,
    id: 1,
    rating: 5,
    user: {
      id: 9,
      name: `Maks`
    },
  },
  {
    comment: `Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2019-30-08T13:38:44.769Z`,
    id: 14,
    rating: 8.3,
    user: {
      id: 3,
      name: `Olga`
    },
  },
];
