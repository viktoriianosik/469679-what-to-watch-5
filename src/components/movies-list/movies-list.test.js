import React from "react";
import renderer from "react-test-renderer";
import {MoviesList} from "./movies-list";
import {testMovies} from "../../test-data/test-movies";

const noop = () => {};

it(`Render component MoviesList correctly`, () => {
  const tree = renderer.create(
      <MoviesList
        movies={testMovies}
        activeMovie={testMovies[0]}
        onCardClick={noop}
        onMoreButtonClick={noop}
        onMouseEnterCard={noop}
        onMouseLeaveCard={noop}
        moviesCount={2}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
