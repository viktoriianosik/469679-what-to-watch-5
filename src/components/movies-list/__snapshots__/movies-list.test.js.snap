// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component MoviesList correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pulp Fiction"
        height="175"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="No Country for Old Men"
        height="175"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/No_Country_for_Old_Men.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
      >
        No Country for Old Men
      </a>
    </h3>
  </article>
</div>
`;
